sklearn/ensemble/forest.py

```diff
@@ -72,17 +73,54 @@ class calls the ``fit`` method of each sub-estimator on random samples
 MAX_INT = np.iinfo(np.int32).max
 
 
-def _generate_sample_indices(random_state, n_samples):
+def _get_n_samples_bootstrap(n_samples, max_samples):
+    """Get the number of samples in a bootstrap sample.
+
+    Parameters
+    ----------
+    n_samples : int
+        Number of samples in the dataset.
+    max_samples : int, float or None (default=None)
+        The maximum number of samples to draw from the total available:
+            - float indicates a fraction of the total.
+            - int indicates the exact number of samples.
+            - None indicates the total number of samples
+
+    Returns
+    -------
+    n_samples_bootstrap : int
+        The total number of samples to draw for the bootstrap sample.
+    """
+    if max_samples is None:
+        return n_samples
+    elif isinstance(max_samples, numbers.Integral):
+        if not (1 <= max_samples <= n_samples):
+            msg = "`max_samples` must be in range 1 to {} but got value {}"
+            raise ValueError(msg.format(n_samples, max_samples))
+        return max_samples
+    elif isinstance(max_samples, numbers.Real):
```

Comment:
I think that pylint will tell you to use an `if` statement here since you have a return or a raise in the previous statement.